### PR TITLE
Refactored hook handlers in mod_amp module

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR changes all hook handlers in `mod_amp` module to `gen_hook` format.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/145218) / small_tests / e0e061bdd532e25037de6d514f63ff173d8e06fa
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3815/145218/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3815/145218/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-18_11.57.59/lib.mongooseim.logs/run.2022-10-18_11.57.59/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/145290) / small_tests / 2b41fb5d0416b43214cb37b9e7c414b2cb168746
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3815/145290/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3815/145290/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-18_12.58.25/lib.mongooseim.logs/run.2022-10-18_12.58.26/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/145374) / small_tests / 6ded2003341e086ddde39e96d30aeeda96e826bb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3815/145374/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3815/145374/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-19_07.00.37/lib.mongooseim.logs/run.2022-10-19_07.00.37/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/145396) / small_tests / 0912341f0ae0895079d34cac83907956c6aad033
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3815/145396/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3815/145396/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-19_07.36.07/lib.mongooseim.logs/run.2022-10-19_07.36.07/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/145860) / small_tests / 80916f89ca1d669b118004333d07ad3f28a60ea5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3815/145860/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3815/145860/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-21_09.55.37/lib.mongooseim.logs/run.2022-10-21_09.55.38/suite.log.html)

File: src/amp_strategy.erl [hunk 1]
```diff
@@ -6,25 +6,25 @@
...
+determine_strategy(_, #{to := To, event := Event}, _) ->
     TargetResources = get_target_resources(To),
     Deliver = deliver_strategy(TargetResources, Event),
     MatchResource = match_resource_strategy(TargetResources),
 
-    #amp_strategy{deliver = Deliver,
+    {ok, #amp_strategy{deliver = Deliver,
                   'match-resource' = MatchResource,
-                  'expire-at' = undefined}.
+                  'expire-at' = undefined}}.
```
Comment (NelsonVides) [inline src/amp_strategy.erl:27]: Update indentation after the previous line got longer :)

**Review by NelsonVides — COMMENTED** [review]
Pretty straightforward, only one minor comment, can merge if you want 🙂

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/146433) / small_tests / badbe55ef2acac881bdc2e104b2fe02e36ec89f4
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3815/146433/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3815/146433/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-24_11.11.19/lib.mongooseim.logs/run.2022-10-24_11.11.20/suite.log.html)

**Review by NelsonVides — APPROVED** [review]
LGTM 👌🏽

**Pull request merged**